File: models/liga.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Liga extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      // Una liga tiene muchas temporadas
      Liga.hasMany(models.Temporada, { foreignKey: 'liga_id' });
    }
  }
  Liga.init({
    nombre: DataTypes.STRING,
    pais: DataTypes.STRING,
    creada_en: DataTypes.DATE
  }, {
    sequelize,
    modelName: 'Liga',
    timestamps: false
  });
  return Liga;
};

File: Flutter/src/nemer_app/lib/models/liga.dart
class Liga {
  final int id;
  final String nombre;
  final String pais;
  final String creadaEn;

  Liga({
    required this.id,
    required this.nombre,
    required this.pais,
    required this.creadaEn,
  });

  factory Liga.fromJson(Map<String, dynamic> json) {
    return Liga(
      id: json['id'] ?? 0,
      nombre: json['nombre'] ?? '',
      pais: json['pais'] ?? '',
      creadaEn: json['creada_en'] ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
        "id": id,
        "nombre": nombre,
        "pais": pais,
        "creada_en": creadaEn,
      };
}
